### ISSUE TITLE
Fix FP S2871 (`no-alphabetical-sort`): Add exception for string arrays

### ISSUE THREAD

**Issue by ilia-kebets-sonarsource** [SonarSource/SonarJS#4531]
Do not raise the issue when the array is an array of strings:

```js
const names = ['foo', 'bar'];
names.sort(); // compliant
```

**Comment by ericmorand-sonarsource** [comment]
@ilia-kebets-sonarsource, wouldn't adding this exception contradict the documentation and the initial purpose of the rule?

I quote the documentation:

> Even to sort strings, the default sort order may give unexpected results. Not only does it not support localization, but it also doesn’t fully support Unicode, as it only considers UTF-16 code units. For example, in the code below, "eΔ" is surprisingly before and after "éΔ".
> ```js
> const code1 = '\u00e9\u0394'; // "éΔ"
> const code2 = '\u0065\u0301\u0394'; // "éΔ" using Unicode combining marks
> const code3 = '\u0065\u0394'; // "eΔ"
> console.log([code1, code2, code3].sort()); // Noncompliant: ["éΔ", "eΔ", "éΔ"], "eΔ" position is inconsistent
> console.log([code1, code2, code3].sort((a, b) => a.localeCompare(b))); // ["eΔ", "éΔ", "éΔ"]
> ```

This case would not trigger the rule anymore if we add the exception. Is it desirable? Should I update the documentation accordingly?

**Comment by ilia-kebets-sonarsource** [comment]
- when we have an array of strings, message: use string.localeCompare(string)
- update RSPEC to make this suggestion more explicit

**Comment by ericmorand-sonarsource** [comment]
@ilia-kebets-sonarsource 

Currently, the message is the same across all cases (arrays of strings, heterogeneous arrays...): "**Provide a compare function to avoid sorting elements alphabetically.**"

Only the suggestions differ. For arrays of strings, the suggestion is "**Add a comparator function to sort in ascending language-sensitive order**."

For arrays of strings, I propose changing the emitted message to something like "_Provide a compare function that depends on string.localeCompare, to reliably sort elements alphabetically._" or "_Provide a compare function that reliably sorts elements alphabetically._" - ideas more than welcome.

The suggestion already looks good to me.

**Comment by ericmorand-sonarsource** [comment]
Successfully checked on Next.

![image](https://github.com/SonarSource/SonarJS/assets/156682586/816168d9-1257-4086-924d-aedab480cc64)
